vertx-core/src/main/java/io/vertx/core/net/impl/ConnectionBase.java

```diff
@@ -399,6 +401,11 @@ public boolean isTrafficShaped() {
 
   public SSLSession sslSession() {
     ChannelHandlerContext sslHandlerContext = chctx.pipeline().context(SslHandler.class);
+    if (sslHandlerContext == null) {
```

Comment:
instead we should use inheritance for that and override in `Http3ConnectionImpl`